src/training/flava_eval/glue2.py

```diff
@@ -0,0 +1,324 @@
+
+import open_clip
+from open_clip.factory import get_tokenizer
+from training.scheduler import cosine_lr
+from training.train import AverageMeter
+
+try:
+    import evaluate
+except ImportError:
+    raise ImportError("Please install HF evaluate: pip install evaluate")
+
+MULTI_PROMPT_TASKS = {"mnli", "mrpc", "qnli", "qqp", "rte", "stsb"}
+TEXT_KEYS = defaultdict(lambda: ("sentence",),
+                    {"mnli": ("premise", "hypothesis"),
+                     "mrpc": ("sentence1", "sentence2"),
+                     "qnli": ("question", "sentence"),
+                     "qqp": ("question1", "question2"),
+                     "rte": ("sentence1", "sentence2"),
+                     "stsb": ("sentence1", "sentence2")})
+
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--task-name",
+        type=str,
+        default=None,
+        help="GLUE task name.",
+    )
+    parser.add_argument(
+        "--workers", type=int, default=4, help="Number of dataloader workers per GPU."
+    )
+    parser.add_argument(
+        "--batch-size", type=int, default=32, help="Batch size per GPU."
+    )
+    parser.add_argument(
+        "--epochs", type=int, default=1000, help="Number of epochs to train for."
+    )
+    parser.add_argument("--lr", type=float, default=3e-5, help="Learning rate.")
+    parser.add_argument("--beta1", type=float, default=0.9, help="Adam beta 1.")
+    parser.add_argument("--beta2", type=float, default=0.999, help="Adam beta 2.")
+    parser.add_argument("--eps", type=float, default=1e-8, help="Adam epsilon.")
+    parser.add_argument("--wd", type=float, default=0.0, help="Weight decay.")
+    parser.add_argument(
+        "--warmup", type=int, default=1000, help="Number of steps to warmup for."
+    )
+    parser.add_argument(
+        "--val-frequency", type=int, default=100, help="How often to run evaluation with val data."
+    )
+    parser.add_argument(
+        "--early-stop-patience", type=int, default=5, help="Early stopping patience."
+    )
+    parser.add_argument(
+        "--early-stop-threshold", type=float, default=0.0, help="Early stopping threshold."
+    )
+    parser.add_argument(
+        "--early-stop-metric-name", type=str, default="accuracy", help="Early stopping metric name."
+    )
+    parser.add_argument(
+        "--precision",
+        choices=["amp", "amp_bf16", "amp_bfloat16", "bf16", "fp16", "fp32"],
+        default="amp",
+        help="Floating point precision."
+    )
+    parser.add_argument(
+        "--model",
+        type=str,
+        default="RN50",
+        help="Name of the vision backbone to use.",
+    )
+    parser.add_argument(
+        "--pretrained",
+        default='',
+        type=str,
+        help="Use a pretrained CLIP model weights with the specified tag or file path.",
+    )
+    parser.add_argument(
+        "--seed", type=int, default=0, help="Default random seed."
+    )
+    parser.add_argument(
```

Comment:
can this be inferred directly from the tokenizer?

Reply:
maybe push in a later PR?